tests/qos/qos_sai_base.py

```diff
@@ -1070,3 +1053,295 @@ def resetWatermark(
+
+            for port in a2.keys():
+                rx_diff = int(a2[port]["RX_OK"]) - int(a1[port]["RX_OK"])
+
+                if rx_diff >= diff:
+                    pytest_assert(
+                        rx_port is None,
+                        "Multiple rx ports with {} rx packets".format(diff)
+                    )
+                    rx_port = port
+
+                tx_diff = int(a2[port]["TX_OK"]) - int(a1[port]["TX_OK"])
+                if tx_diff >= diff:
+                    pytest_assert(
+                        tx_port is None,
+                        "Multiple tx ports with {} tx packets".format(diff)
+                    )
+                    tx_port = port
+
+            # return rx, tx ports that have a packet count difference of > diff
+            return rx_port, tx_port
+
+        test_params["count"] = 100
+        duthost.command("sonic-clear counters")
+        cnt_before = duthost.show_interface(
+            command="counter", asic_index="all", include_internal_intfs=True
+        )
+        # send a burst of packets from a given src IP to dst IP
+        self.runPtfTest(
+            ptfhost, testCase="sai_qos_tests.PacketTransmit",
+            testParams=test_params
+        )
+        time.sleep(8)
+        cnt_after = duthost.show_interface(
+            command="counter", asic_index="all", include_internal_intfs=True
+        )
+
+        asic_idx = None
+        rx_port = None
+        tx_port = None
+
+        # identify the backend ASIC and the rx, tx ports on that ASIC
+        # that forwarded the traffic
+        for asic in duthost.get_backend_asic_ids():
+            rx, tx = find_traffic_ports(
+                asic, cnt_before, cnt_after, test_params["count"]
+            )
+            if rx and tx:
+                pytest_assert(
+                    rx_port is None and tx_port is None,
+                    "Multiple backend ASICs with rx/tx ports"
+                )
+                rx_port, tx_port, asic_idx  = rx, tx, asic
+
+        pytest_assert(asic_idx is not None, "ASIC, rx and tx ports not found")
+        return ({
+            "test_src_port_name": rx_port,
+            "test_dst_port_name": tx_port,
+            "asic_under_test": asic_idx,
+            }
+        )
+
+    @pytest.fixture(scope='class')
+    def build_ip_interface(
+        self, duthosts, rand_one_dut_hostname, swapSyncd, tbinfo
+    ):
+        """
+        builds a list of active IP interfaces and port index
+        for each ASIC
+
+        Returns:
+        {
+            asic_index: {
+                portIndex: {
+                    "ipv4": peer ipv4,
+                    "bgp_neighbor": BGP neighbor
+                }
+                .
+                .
+            }
+           .
+           .
+        }
+        """
+        duthost = duthosts[rand_one_dut_hostname]
+
+        topo = tbinfo["topo"]["name"]
+        if topo not in self.SUPPORTED_T1_TOPOS:
```

Comment:
Is this fixture used only for multi asic platform ?

Reply:
yes, may be there should be a fixture to bail out when not on MASIC platform.